frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx

```diff
@@ -125,8 +131,25 @@ const CellTypeButton = ({
   return (
     <FlexRowJustified data-test-id="cell-type-label-count">
       <FlexRow>
-        <CellTypeLabelStyle data-test-id="cell-type-name">
-          {name}
+        <CellTypeLabelStyle>
+          <Tooltip
+            title={
+              // Set tooltip content only if name is truncated
+              formattedName.includes("...") ? (
+                <CellTypeLabelTooltipStyle>{name}</CellTypeLabelTooltipStyle>
+              ) : null
+            }
+            sdsStyle="light"
+            arrow
+            placement="left"
+            enterNextDelay={700}
+          >
+            {/* Must be wrapped in div and not fragment or else tooltip content won't render */}
+            <div>
+              <HiddenCellTypeLabelStyle>{name}</HiddenCellTypeLabelStyle>
+              <div data-test-id="cell-type-name">{formattedName}</div>
```

Comment:
Can we put this `data-test-id` on `HiddenCellTypeLabelStyle`? I can imagine a situation where we use this ID to select a cell type with a specific name. In that case, we want to use the full (not truncated) name.

Reply:
Alternatively, we could update `data-test-id` to be `cell-type-name-${name}`, but the e2e test selector for all cell types will need to be updated (e.g. like what [this line](https://github.com/chanzuckerberg/single-cell-data-portal/blob/25b4236e7d466f25a0a91cc4c024737fcc76dc7a/frontend/tests/features/wheresMyGene.test.ts#L14) is doing. So, something like `const CELL_TYPE_LABELS_ID = "[data-test-id^=cell-type-name-]";`